### PR TITLE
feat: add possibility to offset CIDRs

### PR DESCRIPTION
**Pull request by argoyle**

### CODE REVIEW

File: vars.tf [hunk 1]
```diff
@@ -13,11 +13,17 @@ variable "vpc_cidr" {
   default     = "172.20.0.0/16"
   description = "CIDR range for the VPC?"
   validation {
-    condition     = cidrsubnet(var.vpc_cidr, 3, 5) != ""
+    condition     = cidrsubnet(var.vpc_cidr, 3, 5 + var.subnet_cidrs_offset) != ""
     error_message = "A larger CIDR range must be provided."
   }
 }
 
+variable "subnet_cidrs_offset" {
```
Comment (argoyle) [inline vars.tf:21]: A better name for this might be better but I can't manage to come up with something better right now 😄

File: vars.tf [hunk 2]
```diff
@@ -13,11 +13,17 @@ variable "vpc_cidr" {
   default     = "172.20.0.0/16"
   description = "CIDR range for the VPC?"
   validation {
-    condition     = cidrsubnet(var.vpc_cidr, 3, 5) != ""
+    condition     = cidrsubnet(var.vpc_cidr, 3, 5 + var.subnet_cidrs_offset) != ""
```
Comment (argoyle) [inline vars.tf:16]: Hmmm, this kind of validation is not possible. Only the current variable can be used. Any thoughts on how to achieve the validation correctly?

File: vars.tf [hunk 3]
```diff
@@ -18,6 +18,12 @@ variable "vpc_cidr" {
   }
 }
 
+variable "subnet_cidrs_offset" {
+  type        = number
+  default     = 1
```
Comment (peter-svensson) [inline vars.tf:23]: Why `1`? 
`0` would give the same as before right?

File: vars.tf (hunk 1 above)
Comment (peter-svensson) [reply vars.tf:21]: Might even be better to allow passing the CIDR ranges as `var`s ?

File: vars.tf (hunk 3 above)
Comment (argoyle) [reply vars.tf]: My mistake. 😊 Fixed!

File: vars.tf (hunk 1 above)
Comment (argoyle) [reply vars.tf:21]: Thought about that as well. The offset worked for the two cases I had but perhaps a map from string to CIDR with overrides is better? We need one for private and one for public in that case, right?

Comment (argoyle) [reply vars.tf]: Ok, added public/private_subnet_cidrs instead.

**Comment by peter-svensson** [comment]
Replaced by #2

**Pull request closed without merging**